api/utils/grpc/interceptors/errors.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package interceptors
+
+import (
+	"context"
+
+	"github.com/gravitational/trace/trail"
+	"google.golang.org/grpc"
+)
+
+// grpcServerStreamWrapper wraps around the embedded grpc.ServerStream
+// and intercepts the RecvMsg and SendMsg method calls converting errors to the
+// appropriate grpc status error.
+type grpcServerStreamWrapper struct {
+	grpc.ServerStream
+}
+
+// SendMsg wraps around ServerStream.SendMsg and adds metrics reporting
+func (s *grpcServerStreamWrapper) SendMsg(m interface{}) error {
+	return trail.FromGRPC(s.ServerStream.SendMsg(m))
+}
+
+// RecvMsg wraps around ServerStream.RecvMsg and adds metrics reporting
+func (s *grpcServerStreamWrapper) RecvMsg(m interface{}) error {
+	return trail.FromGRPC(s.ServerStream.RecvMsg(m))
+}
+
+// grpcClientStreamWrapper wraps around the embedded grpc.ClientStream
+// and intercepts the RecvMsg and SendMsg method calls converting errors to the
+// appropriate grpc status error.
+type grpcClientStreamWrapper struct {
+	grpc.ClientStream
+}
+
+// SendMsg wraps around ClientStream.SendMsg
+func (s *grpcClientStreamWrapper) SendMsg(m interface{}) error {
+	return trail.FromGRPC(s.ClientStream.SendMsg(m))
+}
+
+// RecvMsg wraps around ClientStream.RecvMsg
+func (s *grpcClientStreamWrapper) RecvMsg(m interface{}) error {
+	return trail.FromGRPC(s.ClientStream.RecvMsg(m))
+}
+
+// GRPCServerUnaryErrorInterceptor is a GPRC unary server interceptor that
```

Comment:
```suggestion
// GRPCServerUnaryErrorInterceptor is a GRPC unary server interceptor that
```

Same for others.

Reply:
FWIW this typo wasn't me (for once), it was already there. I went ahead and fixed the other 10 instances of the typo across the codebase.

Reply:
Thanks for fixing it anyway!